### DISCUSSION TITLE
How to dynamically determine the measuring unit of 3D objects' dimensions?

### DISCUSSION THREAD

**Discussion by lemorage** [google/model-viewer#4814] [Q&A] [answered]
### Description

I am referring to the dimensions section in the annotations chapter of the examples part in model-viewer. As the comment shows in the following code cell, the measuring unit is hard-coded by writing `cm` here. But now i have the case that i have a lot of different sizes of 3D objects (some are in meters, and some are in millimeters) to be put on the website, so i am wondering if there is a way to determine the measuring unit based on our objects dynamically? Otherwise, it seems not possible to get the dimensions of different objects in the right measurement unit.
```js
  modelViewer.addEventListener('load', () => {
    const center = modelViewer.getBoundingBoxCenter();
    const size = modelViewer.getDimensions();
    const x2 = size.x / 2;
    const y2 = size.y / 2;
    const z2 = size.z / 2;

    modelViewer.updateHotspot({
      name: 'hotspot-dot+X-Y+Z',
      position: `${center.x + x2} ${center.y - y2} ${center.z + z2}`
    });

    modelViewer.updateHotspot({
      name: 'hotspot-dim+X-Y',
      position: `${center.x + x2 * 1.2} ${center.y - y2 * 1.1} ${center.z}`
    });
    modelViewer.querySelector('button[slot="hotspot-dim+X-Y"]').textContent =
        `${(size.z * 100).toFixed(0)} cm`; // this part

    modelViewer.updateHotspot({
      name: 'hotspot-dot+X-Y-Z',
      position: `${center.x + x2} ${center.y - y2} ${center.z - z2}`
    });

    modelViewer.updateHotspot({
      name: 'hotspot-dim+X-Z',
      position: `${center.x + x2 * 1.2} ${center.y} ${center.z - z2 * 1.2}`
    });
    modelViewer.querySelector('button[slot="hotspot-dim+X-Z"]').textContent =
        `${(size.y * 100).toFixed(0)} cm`; // this part

    // ......
  });
```

#### Live Demo

https://modelviewer.dev/examples/annotations/index.html#dimensions

### Version

- model-viewer: v3.5.0

**Comment by elalish** [comment, marked as answer]
The units in glTF are meters (per the spec), so our sizes are always returned in meters. If your glTF is in any other units, it was exported incorrectly and you'll notice the problem if you go into AR mode where physical scale matters. You may be able to guess the incorrect units (usually when things are 100-1000x too big) and fix them using our `scale` attribute if necessary.

**Reply by lemorage** [reply]
Ok, thank you so much for the answer! So per my understanding, if our model is exported correctly, the physical dimensions should be the same as the dimension data measured by our program (units are meters). but why does the "show-dimensions" example program write all the measuring units in centimeters rather than in meters?

**Reply by elalish** [reply]
Because we felt that most products' dimensions would be clearer to users in `cm`, though in the US it would probably be better to convert to inches for display.

**Reply by lemorage** [reply]
Perfect, that makes much sense. Thank u for your answer!
